fem/CMakeLists.txt

```diff
@@ -19,6 +19,8 @@ set(SRCS
   bilininteg_mass.cpp
   bilininteg_vecdiffusion.cpp
   bilininteg_vecmass.cpp
+  bilininteg_convection.cpp
```

Comment:
How does this PR combines with the previous `convection` one?

Reply:
I think these kernels are more optimized, feel free replace the previous ones. However; I would like if we can maintain the same capabilities in particular ability to handle variable coefficients.

Reply:
There is a big merge conflict, I'll solve it today.

Reply:
The kernels are surprisingly similar, there is very little difference between the your kernels and my unoptimized kernels. I was having `out of memory` problems in 3D though, were you facing similar problems?

Reply:
If that's okay with you I'll remove the case `Q==NULL` where you set the velocity field to `(1,1,1)`, that doesn't really make sense to me. That makes sense for a scalar field to set it to `1`, but not for a vectorial field.